lib/pos_blockchain.go

```diff
@@ -202,8 +202,33 @@ func (bc *Blockchain) validateBlockHeight(desoBlock *MsgDeSoBlock) error {
 // timeout QC, it will check that the view is at least greater than the latest
 // uncommitted block's view + 1.
 func (bc *Blockchain) validateBlockView(desoBlock *MsgDeSoBlock) error {
-	// TODO: Implement me
-	return errors.New("IMPLEMENT ME")
+	committedBlock, _ := bc.getHighestCommittedBlock()
+	if committedBlock == nil {
+		// This is an edge case we'll never hit in practice since all the PoW blocks
+		// are committed.
+		return errors.New("validateBlockView: No committed blocks found")
+	}
+	// Validate that the view is greater than the latest committed block view.
+	if desoBlock.Header.ProposedInView <= committedBlock.Header.ProposedInView {
+		return RuleErrorPoSBlockViewEarlierThanCommittedBlock
+	}
+	// Validate that the view is greater than the latest uncommitted block.
+	chainTip := bc.GetBestChainTip()
+	// If the chain tip is committed, we can just check that the view is greater than the
+	// committed block's view, but this is already kinda done above. Basically we should never
+	// hit this logic.
+	if chainTip.CommittedStatus == COMMITTED {
+		if desoBlock.Header.ProposedInView <= chainTip.Header.ProposedInView {
+			return RuleErrorPoSBlockViewEarlierThanCommittedBlock
+		}
+		return nil
+	}
+	// If the chain tip is uncommitted, we need to check that the view is greater than the
+	// uncommitted block's view.
+	if desoBlock.Header.ProposedInView <= chainTip.Header.ProposedInView {
+		return RuleErrorPoSBlockViewEarlierThanUncommittedBlock
+	}
+	return nil
```

Comment:
I think you need to compare the block's view against its parent's view. If the block has a regular QC, the view needs to be the parent's view + 1. If it's a timeout QC, then the view must be strictly greater than the parent's view + 1.

The block may be part of a fork that doesn't extend from the uncommitted chain tip, so we need to validate the view somehow in that scenario. LMK if I just misunderstood this PR and you plan on doing this separately elsewhere.